lib/src/source_visitor.dart

```diff
@@ -2777,18 +2885,30 @@ class SourceVisitor extends ThrowingAstVisitor {
     token(leftBracket);
     rule.beforeArgument(zeroSplit());
 
+    // Set the block nesting in case an argument is a function type with a
+    // trailing comma or a record type.
+    builder.startBlockArgumentNesting();
+
     for (var node in nodes) {
       visit(node);
 
-      // Write the trailing comma.
+      // Write the comma separator.
       if (node != nodes.last) {
-        token(node.endToken.next);
+        var comma = node.endToken.next;
+
+        // TODO(rnystrom): There is a bug in analyzer where the end token of a
+        // nullable record type is the ")" and not the "?". This works around
+        // that. Remove that's fixed.
+        if (comma?.lexeme == '?') comma = comma?.next;
```

Comment:
I have a fix out for this already: https://dart-review.googlesource.com/c/sdk/+/278696

But the workaround it harmless and this way I can land before waiting for the analyzer folks to kick out a release.